lib/operations/list_collections.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const CommandOperationV2 = require('./command_v2');
+const Aspect = require('./operation').Aspect;
+const defineAspects = require('./operation').defineAspects;
+const maxWireVersion = require('../core/utils').maxWireVersion;
+const listCollectionsTransforms = require('./db_ops').listCollectionsTransforms;
+const CONSTANTS = require('../constants');
+
+const LIST_COLLECTIONS_WIRE_VERSION = 3;
+
+class ListCollectionsOperation extends CommandOperationV2 {
+  constructor(db, filter, options) {
+    super(db, options);
+    this.options.full = true;
```

Comment:
why are we doing this?

Reply:
It's being used [here](https://github.com/mongodb/node-mongodb-native/blob/next/lib/operations/command_v2.js#L38) to inform the base class whether it should return a complete server response, or the `result` field of the response. All methods that are used with cursors require a full response. 

This actually doesn't need to be communicated through `options`, I'll change all of these to not use `options` in a follow-up PR.

Reply:
https://jira.mongodb.org/browse/NODE-1744